src/MoveUntilTouchController.cpp

```diff
@@ -136,31 +120,23 @@ void MoveUntilTouchController::update(const ros::Time& time,
 bool MoveUntilTouchController::shouldAcceptRequests() { return isRunning(); }
 
 //=============================================================================
-bool MoveUntilTouchController::shouldStopExecution()
+bool MoveUntilTouchController::shouldStopExecution(std::string& reason)
 {
   // inelegent to just terminate any running trajectory,
   // but we must guarantee taring completes before starting
   if (!mTaringCompleted.load()) {
     return true;
   }
 
-  const double* fArray = mForceTorqueHandle.getForce();
-  if (fArray == nullptr) {
-    return true;
-  }
-  Eigen::Map<const Eigen::Vector3d> force{fArray};
-
-  const double* tArray = mForceTorqueHandle.getTorque();
-  if (tArray == nullptr) {
-    return true;
-  }
-  Eigen::Map<const Eigen::Vector3d> torque{tArray};
-
   double forceThreshold = mForceThreshold.load();
   double torqueThreshold = mTorqueThreshold.load();
 
-  bool forceThresholdExceeded = force.norm() >= forceThreshold;
-  bool torqueThresholdExceeded = torque.norm() >= torqueThreshold;
+  forceTorqueDataMutex.lock();
+  bool forceThresholdExceeded = mForce.norm() >= forceThreshold;
+  bool torqueThresholdExceeded = mTorque.norm() >= torqueThreshold;
+  forceTorqueDataMutex.unlock();
+
+  reason = "Force or Torque Threshold exceeded!";
```

Comment:
Nit: I think the reason should be more descriptive (which threshold was exceeded, and perhaps by how much?)